frontend/src/upload.js

```diff
@@ -200,7 +345,9 @@ $(document).ready(function() {
             populateFileList(localStorage.getItem(id));
           }
         } else if (xhr.status === 404) {
-          localStorage.removeItem(id);
+          if (isFile(id)) {
```

Comment:
we shouldn't even be calling `checkExistence` on non-file ids. move this up a level